runtime/oti/ObjectHash.hpp

```diff
@@ -230,12 +230,12 @@ class VM_ObjectHash
 						if (0 != offset) {
 							/* Contiguous array */
 							J9ROMArrayClass *romClass = (J9ROMArrayClass*)objectClass->romClass;
```

Comment:
Think if this two blocks between Cont and Discont can be better shared. Chances are not, though, due to deeply nested zero-sized case, that is dependable on header shape (size field offset) between these 2 cases.

Reply:
It is not very sharable for the reasons you have mentioned. I have looked at two different ways of doing it but both don't really improve code sharing and are less readable than what is currently there. However, I will pull out the getting of the contiguous hashcode into its own function which will improve code sharing and readability.